### PR TITLE
[WFLY-13396] Tests for RESTEasy integration with WildFly Elytron - Au…

### PR DESCRIPTION
**Pull request by skyllarr**
…thenticationClient for Authentication / SSL

[WFLY-13396](https://issues.redhat.com/browse/WFLY-13396)

Requires:
[RESTEASY-2366](https://issues.redhat.com/browse/RESTEASY-2366)
[ELY-1891](https://issues.redhat.com/browse/ELY-1891)

### CODE REVIEW

**Comment by Skyllarr** [comment]
retest this please

File: testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/ssl/UndertowTwoWaySslNeedClientAuthTestCase.java [hunk 1]
```diff
@@ -130,6 +146,153 @@ public void testSendingNoClientCertificateFails() {
...
+        });
+        AuthenticationContext contextWithTruststore = doPrivileged((PrivilegedAction<AuthenticationContext>) () -> {
+            try {
+                URL config = getClass().getResource("wildfly-config-correct-truststore.xml");
+                return ElytronXmlParser.parseAuthenticationClientConfiguration(config.toURI()).create();
+            } catch (Throwable t) {
+                throw new InvalidAuthenticationConfigurationException(t);
+            }
+        });
+        SSLContext sslContext = AUTH_CONTEXT_CLIENT.getSSLContext(new URI(securedRootUrl.getHost()), contextWithTruststore);
```
Comment (bstansberry) [inline testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/ssl/UndertowTwoWaySslNeedClientAuthTestCase.java:218]: The value returned from URL.getHost() is not valid input to new URI(String str).

**Review by bstansberry — CHANGES_REQUESTED** [review]
Hi @Skyllarr 

As both of these tests aren't passing on multiple jobs, please be sure they are passing locally for you before using /retest or pushing your branch. Our CI is somewhat overloaded now and this week it is important to reserve it for things that we know will pass. Please be sure they pass in an execution with -Delytron set and another with -Dts.layers set.

I'm going to cancel the jobs currently running against this as we know some are failing.

**Comment by Skyllarr** [comment]
> Hi @Skyllarr
> 
> As both of these tests aren't passing on multiple jobs, please be sure they are passing locally for you before using /retest or pushing your branch. Our CI is somewhat overloaded now and this week it is important to reserve it for things that we know will pass. Please be sure they pass in an execution with -Delytron set and another with -Dts.layers set.
> 
> I'm going to cancel the jobs currently running against this as we know some are failing.

Hi @bstansberry, sorry for me overusing the CI. The tests were passing for both me and QE locally, but were failing on CI. I think there was a problem with my using of hard coded "127.0.0.1" address in those failing tests. I have fixed it and I think the CI should pass now.

**Comment by bstansberry** [comment]
@Skyllarr Thanks. Yes, IPv6 can be an issue that developers don't hit locally. Which is why we make sure CI uses it. :)

Normally CI load isn't a big issue but this week it is.

**Pull request merged**